cat/rc11.cat

```diff
@@ -0,0 +1,29 @@
+RC11
+
+(*
+ * Repaired C11 codel, following
+ * "Repairing Sequential Consistency in C/C++11" in PLDI 2017  , by
+ * Ori Lahav, Viktor Vafeiadis, Jeehoon Kang, Chung-Kil Hur, and Derek Dreyer
+ * Cat coding by Sicon Colin.
+ *)
+
+let myrmw = [RMW] | rmw
+
+let rb = (rf^-1; co) \ id
```

Comment:
I don't really understand this comment. Are you talking about the definition of `myrmw` or `rb`? Why do you think it assumes that RMW is a single event? The model works with herd7 which in the same way as we do, models RMW as pair of events

Reply:
`rb`, primarily,  but `myrmw` might also just be replaced by `rmw`.

Reply:
I think he refers to the fact that `rb` equals `fr \ id`. The question is why is `id` excluded, since only operations that are both read and write at the same time can be connected to themselves via `fr`. This hints at the fact that rmw operations produce a single hybrid event instead of two. But this somewhat contradicts the definition `let myrmw = [RMW] | rmw` which has a binary `rmw`-relation.
Maybe the model was created to work with both, rmw operations modelled as single hybrid event and as two events.

Reply:
Well, we use 2 events, so those 2 adapted definitions of `rmw` and `fr` literally do nothing. While it is true that `myrmw` is larger than `rmw`, the extra elements make no difference at the place where `myrmw` is used. `fr` and `rb` on the other hand are completely identical and so using just `fr` is a no-brainer here.